core/server/proxy/src/main/java/alluxio/proxy/s3/signature/StringToSignProducer.java

```diff
@@ -0,0 +1,326 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.proxy.s3.signature;
+
+import static alluxio.proxy.s3.signature.SignerConstants.TIME_FORMATTER;
+import static alluxio.proxy.s3.signature.SignerConstants.X_AMZ_CONTENT_SHA256;
+import static alluxio.proxy.s3.signature.SignerConstants.X_AMZ_DATE;
+
+import alluxio.proxy.s3.S3ErrorCode;
+import alluxio.proxy.s3.S3Exception;
+import alluxio.proxy.s3.signature.AwsSignatureProcessor.LowerCaseKeyStringMap;
+
+import org.apache.kerby.util.Hex;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.core.MultivaluedMap;
+import java.io.UnsupportedEncodingException;
+import java.net.InetAddress;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URLEncoder;
+import java.net.UnknownHostException;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.StringJoiner;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.time.temporal.ChronoUnit;
+
+/**
+ * Stateless utility to create stringToSign, the base of the signature.
+ */
+public final class StringToSignProducer {
+  private static final Logger LOG = LoggerFactory.getLogger(StringToSignProducer.class);
+  private static final Charset UTF_8 = StandardCharsets.UTF_8;
```

Comment:
See comment about using a reference to an `S3Constant` defining this `Charset`

Reply:
finish.